src/pkg/services/m365/api/lists_pager.go

```diff
@@ -0,0 +1,455 @@
+func (c Lists) GetContentTypes(
+	ctx context.Context,
+	siteID string,
+	listID string,
+	cc CallConfig,
+) ([]models.ContentTypeable, error) {
+	pager := c.NewContentTypesPager(siteID, listID, cc)
+	items, err := pagers.BatchEnumerateItems[models.ContentTypeable](ctx, pager)
+
+	return items, graph.Stack(ctx, err).OrNil()
+}
+
+// ---------------------------------------------------------------------------
+// content types columns pager
+// ---------------------------------------------------------------------------
+
+var _ pagers.NonDeltaHandler[models.ColumnDefinitionable] = &cTypesColumnsPageCtrl{}
+
+type cTypesColumnsPageCtrl struct {
+	siteID  string
+	listID  string
+	cTypeID string
+	gs      graph.Servicer
+	builder *sites.ItemListsItemContentTypesItemColumnsRequestBuilder
+	options *sites.ItemListsItemContentTypesItemColumnsRequestBuilderGetRequestConfiguration
+}
+
+func (p *cTypesColumnsPageCtrl) SetNextLink(nextLink string) {
+	p.builder = sites.NewItemListsItemContentTypesItemColumnsRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *cTypesColumnsPageCtrl) GetPage(
+	ctx context.Context,
+) (pagers.NextLinkValuer[models.ColumnDefinitionable], error) {
+	resp, err := p.builder.Get(ctx, p.options)
+	return resp, graph.Stack(ctx, err).OrNil()
+}
+
+func (p *cTypesColumnsPageCtrl) ValidModTimes() bool {
+	return true
+}
+
+func (c Lists) NewCTypesColumnsPager(
+	siteID string,
+	listID string,
+	cTypeID string,
+	cc CallConfig,
+) *cTypesColumnsPageCtrl {
+	builder := c.Stable.
+		Client().
+		Sites().
+		BySiteId(siteID).
+		Lists().
+		ByListId(listID).
+		ContentTypes().
+		ByContentTypeId(cTypeID).
+		Columns()
+
+	options := &sites.ItemListsItemContentTypesItemColumnsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &sites.ItemListsItemContentTypesItemColumnsRequestBuilderGetQueryParameters{},
+		Headers:         newPreferHeaders(preferPageSize(maxNonDeltaPageSize)),
+	}
+
+	if len(cc.Select) > 0 {
+		options.QueryParameters.Select = cc.Select
+	}
+
+	return &cTypesColumnsPageCtrl{
+		siteID:  siteID,
+		listID:  listID,
+		cTypeID: cTypeID,
+		builder: builder,
+		gs:      c.Stable,
+		options: options,
+	}
+}
+
+// GetCTypesColumns fetches all columns in the content type.
+func (c Lists) GetCTypesColumns(
+	ctx context.Context,
+	siteID string,
+	listID string,
+	cTypeID string,
+	cc CallConfig,
+) ([]models.ColumnDefinitionable, error) {
+	pager := c.NewCTypesColumnsPager(siteID, listID, cTypeID, cc)
+	items, err := pagers.BatchEnumerateItems[models.ColumnDefinitionable](ctx, pager)
+
+	return items, graph.Stack(ctx, err).OrNil()
+}
+
+// ---------------------------------------------------------------------------
+// column links pager
+// ---------------------------------------------------------------------------
+
+var _ pagers.NonDeltaHandler[models.ColumnLinkable] = &columnLinksPageCtrl{}
+
+type columnLinksPageCtrl struct {
+	siteID  string
+	listID  string
+	cTypeID string
```

Comment:
we're trying to avoid abbreviations in var naming, and `cType` isn't easy to grok
```suggestion
	contentTypeID string
```